addons/calendar/controllers/main.py

```diff
@@ -5,61 +5,114 @@
+        ])
+        attendees.do_accept()
+        return self.view_meeting(token, id, attendee.id)
 
     @http.route('/calendar/meeting/decline', type='http', auth="calendar")
-    def decline_meeting(self, token, id, **kwargs):
-        attendee = request.env['calendar.attendee'].sudo().search([
-            ('access_token', '=', token),
-            ('state', '!=', 'declined')])
+    def decline_meeting(self, token, id=False, attendee_id=False, **kwargs):
+        """
+            :param token: str, access token to authenticate the attendee
+            :param id: int, optional, ID of the event
+            :param attendee_id: int, optional, ID of the attendee
+            :param **kwargs: additional keyword arguments for future use
+        """
+
+        attendee = self.fetch_attendee(id, attendee_id, token, 'declined')
+        if not attendee:
+            return request.not_found()
         attendee.do_decline()
-        return self.view_meeting(token, id)
+        return self.view_meeting(token, id, attendee.id)
 
     @http.route('/calendar/recurrence/decline', type='http', auth="calendar")
-    def decline_recurrence(self, token, id, **kwargs):
-        attendee = request.env['calendar.attendee'].sudo().search([
-            ('access_token', '=', token),
-            ('state', '!=', 'declined')])
-        if attendee:
-            attendees = request.env['calendar.attendee'].sudo().search([
-                ('event_id', 'in', attendee.event_id.recurrence_id.calendar_event_ids.ids),
-                ('partner_id', '=', attendee.partner_id.id),
-                ('state', '!=', 'declined'),
-            ])
-            attendees.do_decline()
-        return self.view_meeting(token, id)
+    def decline_recurrence(self, token, id=False, attendee_id=False, **kwargs):
+        """
+            :param token: str, access token to authenticate the attendee
+            :param id: int, optional, ID of the event
+            :param attendee_id: int, optional, ID of the attendee
+            :param **kwargs: additional keyword arguments for future use
+        """
+
+        attendee = self.fetch_attendee(id, attendee_id, token, 'declined')
+        if not attendee:
+            return request.not_found()
+        attendees = request.env['calendar.attendee'].sudo().search([
+            ('event_id', 'in', attendee.event_id.recurrence_id.calendar_event_ids.ids),
+            ('partner_id', '=', attendee.partner_id.id),
+            ('state', '!=', 'declined'),
+        ])
+        attendees.do_decline()
+        return self.view_meeting(token, id, attendee.id)
 
     @http.route('/calendar/meeting/view', type='http', auth="calendar")
-    def view_meeting(self, token, id, **kwargs):
-        attendee = request.env['calendar.attendee'].sudo().search([
-            ('access_token', '=', token),
-            ('event_id', '=', int(id))])
-        if not attendee:
+    def view_meeting(self, token, id, attendee_id=False, **kwargs):
```

Comment:
Do we still need id if we have attendee_id? afaik you should be able to infer it so maybe we just remove it from the route now? (making id optional)